java/src/main/native/src/TableJni.cpp

```diff
@@ -337,25 +340,23 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_gdfLeftJoin(
     context.flag_sort_result = 1;
     context.flag_sort_inplace = 0;
 
-    int result_num_cols =
-        n_left_table->num_columns() + n_right_table->num_columns() - left_join_cols_arr.size();
-
-    // gdf_left_join is allocating the memory for the results so
-    // allocate the output column structures here when we get it back fill in
-    // the the outPtrs
-    cudf::jni::unique_jpointerArray<gdf_column> output_columns(env, result_num_cols);
-    for (int i = 0; i < result_num_cols; i++) {
-      output_columns.reset(i, new gdf_column());
+    int dedupe_size = left_join_cols.size();
+    std::vector<std::pair<gdf_size_type, gdf_size_type>> dedupe(dedupe_size);
+    for (int i = 0; i < dedupe_size; i++) {
+      dedupe[i].first = left_join_cols[i];
+      dedupe[i].second = right_join_cols[i];
```

Comment:
This makes sense to me but there is an example in the API doc that we won't be able to do. 

https://github.com/rapidsai/cudf/blob/12f15c81c36659cdec2935cfec38b02968a4deda/cpp/include/cudf/join.hpp#L90

But it's irrelevant to us

Reply:
The goal of this was to unbreak the build.  We can explore the usefulness of exposing columns_in_common later on if it is needed.

Reply:
👍